### PR TITLE
Cmnist

### PR DESCRIPTION
**Pull request by olliethomas**
I'd like it if CMNIST were in EthicML. I've copied over the code from NoSINN with a small working (or pretty sure it's working at least) example in the tests. Opening a draft PR so that @MylesBartlett will hopefully take a look 🙏

I'll work on mypy / pylint / doctrings in the morning 💤

### CODE REVIEW

File: ethicml/data/load.py [hunk 1]
```diff
@@ -5,11 +5,10 @@
 
 import pandas as pd
 
+from ethicml.data.tabular_data.configurable_dataset import ConfigurableDataset
+from ethicml.data.tabular_data.dataset import Dataset
```
Comment (thomkeh) [inline ethicml/data/load.py:9]: I don't like these absolute imports

File: ethicml/data/__init__.py [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+from .tabular_data.dataset import *
+from .tabular_data.german import *
+from .tabular_data.non_binary_toy import *
+from .tabular_data.sqf import *
+from .tabular_data.toy import *
+from .tabular_data.util import *
+from .vision_data.celeba import *
+from .vision_data.cmnist import *
+from .vision_data.genfaces import *
+from .vision_data.image_dataset import *
```
Comment (thomkeh) [inline ethicml/data/__init__.py:16]: why make this unrelated change in this PR?

File: ethicml/data/csvs/make_health_from_raw.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 import pandas as pd
 
-df = pd.read_csv("./raw/health.csv")
+df = pd.read_csv("raw/health.csv")
```
Comment (thomkeh) [inline ethicml/data/csvs/make_health_from_raw.py:6]: why make it less explicit?

File: ethicml/data/__init__.py (hunk 2 above)
Comment (MylesBartlett) [reply ethicml/data/__init__.py:16]: I would also like to know!

File: ethicml/data/load.py (hunk 1 above)
Comment (MylesBartlett) [reply ethicml/data/load.py:9]: Nor do I!

File: ethicml/data/vision_data/cmnist.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    dataset_size = len(all_data)
+    indices = list(range(dataset_size))
+    split = int(np.floor((1 - test_pcnt) * dataset_size))
+
+    np.random.shuffle(np.array(indices))
+
+    train_indices, test_indices = indices[:split], indices[split:]
+
+    train_data = Subset(all_data, indices=train_indices)
+    test_data = Subset(all_data, indices=test_indices)
```
Comment (thomkeh) [inline ethicml/data/vision_data/cmnist.py:107]: split with `torch.utils.data.random_split`?

File: ethicml/implementations/pytorch_common.py [hunk 5]
```diff
@@ -122,6 +122,29 @@ def pad_second(x):
     return mmd2
 
 
+def linear_time_mmd(x: np.ndarray, y: np.ndarray, sigma: float = 0.1) -> float:
```
Comment (thomkeh) [inline ethicml/implementations/pytorch_common.py:125]: unrelated to PR

File: ethicml/data/vision_data/cmnist.py (hunk 4 above)
Comment (MylesBartlett) [reply ethicml/data/vision_data/cmnist.py:107]: Why not.

File: ethicml/implementations/vfae_modules/vfae_network.py [hunk 6]
```diff
@@ -6,7 +6,7 @@
 import torch
 from torch import Tensor, nn
 
-from ethicml.data.dataset import Dataset
+from ethicml.data.tabular_data.dataset import Dataset
```
Comment (thomkeh) [inline ethicml/implementations/vfae_modules/vfae_network.py:9]:
```suggestion
from ethicml.data import Dataset
```

File: ethicml/implementations/pytorch_common.py (hunk 5 above)
Comment (MylesBartlett) [reply ethicml/implementations/pytorch_common.py:125]: Very

File: ethicml/data/csvs/make_health_from_raw.py (hunk 3 above)
Comment (olliethomas) [reply ethicml/data/csvs/make_health_from_raw.py:6]: tbh I don't know how this happened
